### PR TITLE
Add Gosec Github Action

### PR DESCRIPTION
**Pull request by rauljordan**
This PR adds the gosec Github action which scans for vulnerabilities in Go repositories from a set of nice rules, see [here](https://github.com/securego/gosec) for more details

### CODE REVIEW

**Review by terencechain — APPROVED** [review]

File: tools/specs-checker/download.go [hunk 1]
```diff
@@ -50,7 +50,7 @@ func getAndSaveFile(specDocUrl, outFilePath string) error {
 	}()
 
 	// Download spec doc.
-	resp, err := http.Get(specDocUrl)
+	resp, err := http.Get(specDocUrl) /* #nosec G107 */
```
Comment (prestonvanloon) [inline tools/specs-checker/download.go:53]: Can we fix this? What is the solution for G107?

Comment (prestonvanloon) [reply tools/specs-checker/download.go:53]: G107: Url provided to HTTP request as taint input

Comment (rauljordan) [reply tools/specs-checker/download.go:53]: it doesn't want untrusted, dynamically provided URLs to perform http requests. The resolution is to hardcode the spec url via a constant, or at least the base path perhaps

Comment (rauljordan) [reply tools/specs-checker/download.go:53]: In this case, there is a base path specified in a function above and this function is unexported, so cannot be used maliciously by anyone else

**Review by terencechain — APPROVED** [review]

**Pull request merged**